### PR TITLE
feat:number list style update

### PR DESCRIPTION
**Pull request by q200892907**

### CODE REVIEW

**Comment by LucasXu0** [comment]
@q200892907 Please include the required tests.

**Comment by q200892907** [comment]
> @q200892907 Please include the required tests.

add print test

File: lib/src/editor/block_component/numbered_list_block_component/numbered_list_block_component.dart [hunk 1]
```diff
@@ -253,3 +283,22 @@ class _NumberedListIconBuilder {
...
+    }
+    return result.toLowerCase();
+  }
+
+  String toRomanString() {
+    final ToRoman standardToRoman = ToRoman.standard();
+
+    return standardToRoman(this).toLowerCase();
+  }
+}
```
Comment (LucasXu0) [inline lib/src/editor/block_component/numbered_list_block_component/numbered_list_block_component.dart:304]:
```suggestion
extension on int {
  String get latin {
    String result = '';
    int number = this;
    while (number > 0) {
      int remainder = (number - 1) % 26;
      result = String.fromCharCode(remainder + 65) + result;
      number = (number - 1) ~/ 26;
    }
    return result.toLowerCase();
  }

  String get roman {
    return toRomanNumeralString() ?? '$this';
  }
}
```

File: lib/src/editor/block_component/numbered_list_block_component/numbered_list_block_component.dart [hunk 2]
```diff
@@ -217,14 +217,32 @@ class _NumberedListIcon extends StatelessWidget {
...
+            style: text.combine(textStyle),
+          ),
           textDirection: direction,
         ),
       ),
     );
   }
 }
 
+extension NumberedListNodeExt on Node {
```
Comment (LucasXu0) [inline lib/src/editor/block_component/numbered_list_block_component/numbered_list_block_component.dart:231]:
```dart
extension on Node {
  String get levelString {
    final builder = _NumberedListIconBuilder(node: this);
    final indexInRootLevel = builder.indexInRootLevel;
    final indexInSameLevel = builder.indexInSameLevel;
    final level = indexInRootLevel % 3;
    final levelString = switch (level) {
      1 => indexInSameLevel.latin,
      2 => indexInSameLevel.roman,
      _ => '$indexInSameLevel',
    };
    return '$levelString.';
  }
}

class _NumberedListIconBuilder {
  _NumberedListIconBuilder({
    required this.node,
  });

  final Node node;

  // the level of the current node
  int get indexInRootLevel {
    var level = 0;
    var parent = node.parent;
    while (parent != null) {
      if (parent.type == NumberedListBlockKeys.type) {
        level++;
      }
      parent = parent.parent;
    }
    return level;
  }

  // the index of the current level
  int get indexInSameLevel {
    int level = 1;
    Node? previous = node.previous;

    // if the previous one is not a numbered list, then it is the first one
    if (previous == null || previous.type != NumberedListBlockKeys.type) {
      return node.attributes[NumberedListBlockKeys.number] ?? level;
    }

    int? startNumber;
    while (previous != null && previous.type == NumberedListBlockKeys.type) {
      startNumber = previous.attributes[NumberedListBlockKeys.number] as int?;
      level++;
      previous = previous.previous;
    }
    if (startNumber != null) {
      return startNumber + level - 1;
    }
    return level;
  }
}
```

File: test/editor/block_component/numbered_list_block_component/numbered_list_level_string_test.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import 'dart:convert';
+
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
```
Comment (LucasXu0) [inline test/editor/block_component/numbered_list_block_component/numbered_list_level_string_test.dart:7]:
```dart
    testWidgets('number, latin and roman', (tester) async {
      final delta = Delta()..insert(text);
      final editor = tester.editor
        ..addNode(
          // 1.
          numberedListNode(
            delta: delta,
            children: [
              // a.
              numberedListNode(
                delta: delta,
                children: [
                  numberedListNode(delta: delta), // Ⅰ.
                  numberedListNode(delta: delta), // ⅠⅠ.
                  numberedListNode(delta: delta), // ⅠⅠⅠ.
                ],
              ),
              // b.
              numberedListNode(
                delta: delta,
                children: [
                  numberedListNode(delta: delta), // Ⅰ.
                  numberedListNode(delta: delta), // ⅠⅠ.
                  numberedListNode(delta: delta), // ⅠⅠⅠ.
                ],
              ),
            ],
          ),
        )
        ..addNode(
          // 2.
          numberedListNode(
            delta: delta,
          ),
        );

      await editor.startTesting();
      for (final number in ['1.', '2.']) {
        expect(find.text(number, findRichText: true), findsOneWidget);
      }
      for (final latin in ['a.', 'b.']) {
        expect(find.text(latin, findRichText: true), findsOneWidget);
      }
      for (final roman in ['I.', 'II.', 'III.']) {
        expect(find.text(roman, findRichText: true), findsNWidgets(2));
      }
    });
```

File: test/editor/block_component/numbered_list_block_component/numbered_list_level_string_test.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  group('NumberedList', () {
+    test('LevelString', () async {
+      void printNodeLevel(List<Node> nodes) {
+        for (var element in nodes) {
+          debugPrint(element.getLevelString());
+          if (element.children.isNotEmpty) {
+            printNodeLevel(element.children);
```
Comment (LucasXu0) [inline test/editor/block_component/numbered_list_block_component/numbered_list_level_string_test.dart:14]: It's not stable for testing using print.

**Comment by LucasXu0** [comment]
@q200892907 I have refactored some parts of the code and made the tests more robust.

**Pull request merged**